File: src/app/models/Account.js

// Define collection and schema for Account
export interface Account {
    accountNumber:
	type : AccountNumber
    iban:
	type : IBAN
    accountName:
	type : string
    currency:
	type : string
    openedOn:
	type : Date
    closedOn:
	type : Date
    Bank:
	type : Schema.Types.ObjectId
    Branch:
	type : Schema.Types.ObjectId
    Product:
	type : Schema.Types.ObjectId
    Owners:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Customer' }]
    Transactions:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Transaction' }]
    Statements:
 	type : [{ type: Schema.Types.ObjectId, ref: 'AccountStatement' }]
    StandingInstructions:
 	type : [{ type: Schema.Types.ObjectId, ref: 'StandingInstruction' }]
    FeeCharges:
 	type : [{ type: Schema.Types.ObjectId, ref: 'FeeCharge' }]
    AccountType:
 	type : String
    OwnershipType:
 	type : String
    Status:
 	type : String
#
    collection: 'accounts'
}

File: bankingOnGolang/internal/model/Account.go
package model

import (
	"time"
)

// ==============================================================
// Account Declaration
// ==============================================================
type Account struct {
	BaseModel
	AccountNumber        AccountNumber
	Iban                 IBAN
	AccountName          string
	Currency             string
	OpenedOn             time.Time
	ClosedOn             time.Time
	BankId               *uint
	Bank                 *Bank `gorm:"foreignKey:BankId"`
	BranchId             *uint
	Branch               *Branch `gorm:"foreignKey:BranchId"`
	ProductId            *uint
	Product              *BankingProduct       `gorm:"foreignKey:ProductId"`
	Owners               []Customer            `gorm:"foreignKey:OwnersFromAccountId"`
	Transactions         []Transaction         `gorm:"foreignKey:TransactionsFromAccountId"`
	Statements           []AccountStatement    `gorm:"foreignKey:StatementsFromAccountId"`
	StandingInstructions []StandingInstruction `gorm:"foreignKey:StandingInstructionsFromAccountId"`
	FeeCharges           []FeeCharge           `gorm:"foreignKey:FeeChargesFromAccountId"`
	AccountType          AccountType
	OwnershipType        AccountOwnershipType
	Status               AccountStatus

	// parent associations as their child

}
